src/components/PeopleFilters.tsx

```diff
@@ -1,12 +1,92 @@
+import { Link, useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { SexFilter, SenturyFilter } from '../types/FilterKeys';
+import { SearchParms } from '../types/SearchParams';
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const handleCenturiesLink = (paramValue: string | null) => {
+    const newSerchParams = new URLSearchParams(searchParams);
+
+    if (!paramValue) {
+      newSerchParams.delete(SearchParms.Centuries);
+
+      return newSerchParams.toString();
+    }
+
+    const centuriesList = newSerchParams.getAll(SearchParms.Centuries);
+
+    if (centuriesList.includes(paramValue)) {
+      newSerchParams.delete(SearchParms.Centuries);
+      centuriesList
+        .filter(cent => cent !== paramValue)
+        .forEach(el => newSerchParams.append(SearchParms.Centuries, el));
```

Comment:
It is not necessary to go through the array twice, it is enough to transfer the condition that is currently in the `filter` method in the `forEach` method